### PR TITLE
fix(upgrade): remove deprecation warnings

### PR DESCRIPTION
**Pull request by obartra**
Upgrade to React 15.5.4 and fixed deprecation warnings.

Closes https://github.com/Hacker0x01/react-datepicker/issues/813

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -90,6 +92,7 @@
   "dependencies": {
     "classnames": "^2.2.5",
     "moment": "^2.17.1",
+    "prop-types": "^15.5.8",
     "react-onclickoutside": "^5.9.0",
```
Comment (sebastiandeutsch) [inline package.json]: @obartra if you want to get rid of all deprecation warnings you'll need to update `react-onclickoutside` to `5.11.1`.

**Comment by sebastiandeutsch** [comment]
@obartra do you need any helping hands with this one? I've seen that you transitioned all the classes in your fork. Anything else to do?

**Comment by obartra** [comment]
I need to take another pass at it. I broke a couple tests in `day.jsx` but should be easy to fix.

My main concern is on making sure we are ok using `transform-class-properties`. It's a stage 2 feature but help us get rid of `.bind` everywhere.

**Comment by sebastiandeutsch** [comment]
Awesome! I'm using `transform-class-properties` in my projects too - but eventually @Hacker0x01 has to decide. Ping me if you need help with tedious tasks like putting the binds in the constructor in countless classes :-)

**Comment by obartra** [comment]
@martijnrusschen can you guys take a look see if this along the lines of what you are looking for?

**Comment by martijnrusschen** [comment]
Looks great! Thanks for the contribution!

**Pull request merged**